File: openmp-hypack-2013/openmp-codes-f-lang/omp-unique-threadid.f


C***************************************************************************
c
c		C-DAC Tech Workshop : hyPACK-2013
c                         October 15-18, 2013
c
C Example 1.1      : omp_unique_threadid.f
C
C Objective        : Write a simple OpenMP program to print unique number for
C                    each thread using the !$OMP PARALLEL directive.
C                    This example demonstrates the use of OpenMP PARALLEL
C                    directive and OMP_GET_THREAD_NUM run-time library routine
C 
C Input            : User has to set OMP_NUM_THREADS environment variable
C	             for n number of threads
C
C Output           : Each thread prints its thread identifier and master
C                    thread prints total number of threads used in the
C              	     program.	                                            
C                                                                        
c   Created        : August-2013
c
c   E-mail         : hpcfte@cdac.in     
c
C*******************************************************************************


       program UniqueThreadID
       integer OMP_GET_THREAD_NUM, OMP_GET_NUM_THREADS
       integer ThreadID, NoofThreads

C$OMP PARALLEL PRIVATE (ThreadID, NoofThreads)

       ThreadID = OMP_GET_THREAD_NUM()
       print *, "My Thread ID is", ThreadID

       if(ThreadID .eq. 0) then
         NoofThreads = OMP_GET_NUM_THREADS()
         print*,"Number of threads = ", NoofThreads
       endif

C$OMP END PARALLEL

       end

File: coprocessor-hypack-2013/coprocessor-xeon-phi-codes/z-mat_all/not-useful-2013/omp-unique-threadid.c


/*****************************************************************************************
			 C-DAC Tech Workshop : HeGaPa-2012
                             July 16-20,2012



 Example 1.1           : omp-unique-threadid.c


 Objective             :  Write a simple OpenMP program to print unique number for
                  	  each thread started by the #pragma parallel.
                  	  This example demonstrates the use of OpenMP PARALLEL
                  	  Directive and omp_get_thread_num() call
 
 Input                 : Set the number of threads to use by means of the OMP_NUM_THREADS environment variable.
		   
		   		For C shell use command :
		
					setenv OMP_NUM_THREADS 4 
	
		   		For bash shell use command :
		
					export OMP_NUM_THREADS=4 
 

 Output                : Each thread prints its thread id.	                                            
                                                                        
 Created               : MAY-2012 
       
 
 E-mail                : betatest@cdac.in                                          


******************************************************************************************/



#include <stdio.h>
#include <stdlib.h>
#include <omp.h>

/* Main Program */
main(int argc,char **argv)
{
	int             threadid,Noofthreads;


 	printf("\n\t\t---------------------------------------------------------------------------");
        printf("\n\t\t Centre for Development of Advanced Computing (C-DAC)");
        printf("\n\t\t Email : betatest@cdac.in");
        printf("\n\t\t---------------------------------------------------------------------------");
        printf("\n\t\t Objective : OpenMP program to print unique thread identifier for ");
	printf("\n\t\t each thread using OpenMP PARALLEL directive.");
        printf("\n\t\t..........................................................................\n");

	printf("\n\n\t\t Master thread  prints this before forking the team of thread \n");

        /* Set the number of threads */
	/*omp_set_num_threads(4);*/

	/* OpenMP Parallel Construct : Fork a team of threads */ 
	#pragma omp parallel private(threadid)
	{
                /* Obtain the thread id */
		threadid = omp_get_thread_num();

		/* Each Thread Prints Its Threadid */
		printf("\n\t\t My thread id is : %d\n", threadid);
	
	} /* All thread join Master thread */

	printf("\n\t\t Master thread  prints this after the end parallel region \n \n");
}
